src/bun.js/node/node_fs_stat_watcher.zig

```diff
@@ -36,45 +36,102 @@ pub const StatWatcherScheduler = struct {
     timer: ?*uws.Timer = null,
 
     head: std.atomic.Value(?*StatWatcher) = .{ .raw = null },
```

Comment:
`UnboundedQueue` should handel the queue of stat watchers and that will simplify this code a lot